core/src/main/java/com/linecorp/armeria/client/endpoint/StickyStrategy.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint;
+
+import java.util.List;
+import java.util.TreeMap;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * An {@code EndpointSelector} strategy which implement sticky load-balancing using
+ * user passed {@code RequestContextHasher} with consistent hashing.
+ */
+public class StickyStrategy implements EndpointSelectionStrategy {
+
+    private final RequestContextHasher requestHasher;
+
+    private final EndpointHasher endpointHasher;
+
+    private static final EndpointHasher DEFAULT_ENDPOINT_HASHER = Endpoint::hashCode;
+
+    /**
+     * An interface represents strategy to hash a ClientRequestContext to a long.
+     */
+    @FunctionalInterface
+    public interface RequestContextHasher {
+        /**
+         * Hash an {@code ClientRequestContext} to a Long.
+         * @param ctx The {@code ClientRequestContext} request context
```

Comment:
The -> the?
How about just `the {@code ClientRequestContext}`? or I think we can remove this line since there's not so much words that explaining the parameter. :-)